R/ggplotMDS.R

```diff
@@ -523,11 +523,31 @@ MDS_var_explained <- function(mds,
         mds <- limma::plotMDS(mds, plot = FALSE)
     }
 
-    mds.distances <- mds %$% distance.matrix %>% as.dist
-    mdsvals <- mds.distances %>%
-        {suppressWarnings(cmdscale(., k = ncol(mds$distance.matrix) - 1))} %>%
-        magrittr::set_colnames(stringr::str_c("Dim", seq_len(ncol(.)))) %>%
-        as.data.frame
+    mds.distances <- tryCatch(
+        expr = {
+            mds %$% distance.matrix %>% as.dist
+        },
+        error = function(e) {
+            mds %$% distance.matrix.squared %>% as.dist
```

Comment:
I tend to not like to use the exposition pipe - is it necessary here or can we write this slightly differently and avoid bringing this pipe in.  If it's already used elsewhere in the code then this is fine, otherwise, if it is newly used I'd prefer we don't use it if possible.

Reply:
this was there before. I don't recall seeing it in other parts of the code. I'll look to see if I can find a way around it.

Reply:
put in a task here - https://3.basecamp.com/3775257/buckets/16965597/todos/4581566494